### PR TITLE
Simplify "interface as resource" test case

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A
| License       | MIT
| Doc PR        | N/A

Actually needed in #2884, but to ease maintenance, this commit is cherry-picked for `2.4`.

### CODE REVIEW

**Comment by teohhanhui** [comment]
AppVeyor failure due to symfony/symfony#32395

File: tests/Fixtures/TestBundle/DataProvider/TaxonItemDataProvider.php [hunk 1]
```diff
@@ -41,6 +44,8 @@ public function supports(string $resourceClass, string $operationName = null, ar
      */
     public function getItem(string $resourceClass, $id, string $operationName = null, array $context = [])
     {
-        return $this->managerRegistry->getRepository(Taxon::class)->find($id);
+        return $this->managerRegistry->getRepository($this->orm ? Taxon::class : TaxonDocument::class)->findOneBy([
```
Comment (alanpoulain) [inline tests/Fixtures/TestBundle/DataProvider/TaxonItemDataProvider.php]: Maybe we can have a `get_class` for the manager registry instead of `$orm`?

Comment (teohhanhui) [reply tests/Fixtures/TestBundle/DataProvider/TaxonItemDataProvider.php]: The problem is, we cannot tell whether we're using the ORM or ODM, as it's the same interface as `$resourceClass` for both.

Comment (teohhanhui) [reply tests/Fixtures/TestBundle/DataProvider/TaxonItemDataProvider.php]: Oh, I get what you mean now. But that seems hacky.

Comment (teohhanhui) [reply tests/Fixtures/TestBundle/DataProvider/TaxonItemDataProvider.php]: And in fact, both are https://github.com/symfony/symfony/blob/v4.3.2/src/Symfony/Bridge/Doctrine/ManagerRegistry.php so it doesn't help.

**Pull request merged**